packages/x-data-grid/src/models/api/gridFilterApi.ts

```diff
@@ -61,4 +62,10 @@ export interface GridFilterApi {
    * Returns the value of the `ignoreDiacritics` prop.
    */
   ignoreDiacritics: DataGridProcessedProps['ignoreDiacritics'];
+  /**
+   * Returns the filter state for the given filter model without applying it to the data grid.
+   * @param {GridFilterModel} filterModel The filter model to get the state for.
+   * @returns {GridStateCommunity['filter']} The filter state.
+   */
+  getFilterState: (filterModel: GridFilterModel) => GridStateCommunity['filter'];
```

Comment:
This sounds like it should be a (pure) selector, but I see how that would be annoying to implement.